pkg/kubectl/polymorphichelpers/canbeautoscaled.go

```diff
@@ -36,7 +39,11 @@ func canBeAutoscaled(kind schema.GroupKind) error {
 		extensionsv1beta1.SchemeGroupVersion.WithKind("ReplicaSet").GroupKind():
 		// nothing to do here
 	default:
-		return fmt.Errorf("cannot autoscale a %v", kind)
+		gr := mapping.GroupVersionKind.GroupVersion().WithResource(mapping.Resource.Resource).GroupResource()
+		_, err := scaler.Scales(info.Namespace).Get(gr, info.Name)
```

Comment:
Discovery information has this as well. Can't we drop all the special case above and just depend on that?

Reply:
/cc @s-urbaniak @DirectXMan12

Reply:
yes, we should be able to

Reply:
Switched to the dynamic client. Does it look like what you had in mind?